lua/build.sh

```diff
@@ -2,34 +2,59 @@
 
 set -e
 
-LUA=lua-5.3.5
-PKG_URL="https://www.lua.org/ftp/${LUA}.tar.gz"
-PKG_MIRROR_URL="https://files.phoesys.com/ports/${LUA}.tar.gz"
+version="${PORTS_LUA_VERSION:-5.3.6}"
+archive_filename="lua-${version}.tar.gz"
+tests_version="5.4.6"
 
-b_log "Building lua"
-PREFIX_LUA="${PREFIX_PROJECT}/phoenix-rtos-ports/lua"
-PREFIX_LUA_BUILD="${PREFIX_BUILD}/lua"
-PREFIX_LUA_SRC="${PREFIX_LUA_BUILD}/${LUA}"
+PREFIX_PORT_SRC="${PREFIX_PORT_BUILD}/${version}"
+PREFIX_PORT_TESTS="${PREFIX_PORT_BUILD}/${tests_version}-tests"
 
+b_port_download "https://www.lua.org/ftp/" "${archive_filename}"
 
-mkdir -p "$PREFIX_LUA_BUILD"
-if [ ! -f "$PREFIX_LUA/${LUA}.tar.gz" ]; then
-	if ! wget "$PKG_URL" -P "${PREFIX_LUA}" --no-check-certificate; then
-		wget "$PKG_MIRROR_URL" -P "${PREFIX_LUA}" --no-check-certificate
-	fi
+if [ ! -d "${PREFIX_PORT_SRC}" ]; then
+	echo "Extracting sources from ${archive_filename}"
+	mkdir -p "${PREFIX_PORT_SRC}"
+	tar -axf "${PREFIX_PORT}/${archive_filename}" --strip-components 1 -C "${PREFIX_PORT_SRC}"
 fi
-if [ ! -d "$PREFIX_LUA_SRC" ]; then
-	tar zxf "$PREFIX_LUA/${LUA}.tar.gz" -C "$PREFIX_LUA_BUILD"
-	cp "$PREFIX_LUA/Makefile" "$PREFIX_LUA_SRC/src/"
+
+b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}"
+
+if [ "${PORTS_LUA_RESTRAIN}" = "y" ]; then
+	b_port_apply_patches "${PREFIX_PORT_SRC}" "${version}/restrain"
 fi
 
+mycflags=(
+	${CFLAGS}
+	-DLUAI_MAXSTACK=${PORTS_LUA_STACK_SIZE:-2000}
```

Comment:
`LUAI_MAXSTACK` passed to lualib should be compatible with the one that is given to applications (as it is used in one of the header files of Lua API). The value of  `PORTS_LUA_STACK_SIZE` will be used outside of ports too.